### PR TITLE
Add support for the experimental wasi-crypto APIs

### PR DESCRIPTION
**Pull request by jedisct1**
Add support for the experimental wasi-crypto APIs.

This new PR imports only the wiggle/wasmtime integration bits into the `wasmtime` repository.

The purpose of the implementation is to allow bindings and application developers to test the proposed APIs. Rust and AssemblyScript bindings are also available as examples.

Like `wasi-nn`, it is currently disabled by default, and requires the `wasi-crypto` feature flag to be compiled in.

### CODE REVIEW

File: crates/wasi-crypto/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+license = "Apache-2.0 WITH LLVM-exception"
+categories = ["wasm", "cryptography"]
+keywords = ["webassembly", "wasm", "crypto"]
+repository = "https://github.com/bytecodealliance/wasmtime"
+readme = "README.md"
+edition = "2018"
+
+[dependencies]
+anyhow = "1.0"
+wasi-crypto = "0.1.3"
```
Comment (alexcrichton) [inline crates/wasi-crypto/Cargo.toml:16]: Is this intended to be a `path` dependency into the submodule perhaps?

**Review by alexcrichton — COMMENTED** [review]
Looks great to me, thanks! Just one question about the dependency though

File: crates/wasi-crypto/Cargo.toml (hunk 1 above)
Comment (jedisct1) [reply crates/wasi-crypto/Cargo.toml]: `path` dependency added!

Comment (jedisct1) [reply crates/wasi-crypto/Cargo.toml]: That requires a change to the `publish` script. Done.

**Review by alexcrichton — APPROVED** [review]

**Pull request merged**